packages/devtools_app/lib/src/screens/performance/timeline_flame_chart.dart

```diff
@@ -46,117 +49,124 @@ class TimelineAnalysisContainer extends StatefulWidget {
-          _controller.data!.eventGroups.isEmpty;
-      if (widget.processing) {
-        content = _buildProcessingInfo();
-      } else if (timelineEmpty) {
-        content = Center(
-          key: TimelineAnalysisContainer.emptyTimelineKey,
-          child: Row(
-            mainAxisAlignment: MainAxisAlignment.center,
-            children: [
-              Text(
-                'No timeline events. Try clicking the refresh button ',
-                style: Theme.of(context).subtleTextStyle,
-              ),
-              Icon(
-                Icons.refresh,
-                size: defaultIconSize,
-              ),
-              Text(
-                ' to load more data.',
-                style: Theme.of(context).subtleTextStyle,
-              ),
-            ],
-          ),
-        );
-      } else {
-        content = LayoutBuilder(
-          builder: (context, constraints) {
-            return TimelineFlameChart(
-              _controller.data!,
-              width: constraints.maxWidth,
-              height: constraints.maxHeight,
-              selectionNotifier: _controller.selectedTimelineEvent,
-              searchMatchesNotifier: _controller.searchMatches,
-              activeSearchMatchNotifier: _controller.activeSearchMatch,
-              onDataSelected: _controller.selectTimelineEvent,
-            );
-          },
-        );
-      }
+      frameAnalysisView = const Center(
+        child: Text('Select a frame above to view analysis data.'),
+      );
     }
 
-    final searchFieldEnabled = selectedTab == null &&
-        !(_controller.data?.isEmpty ?? true) &&
-        !widget.processing;
-
-    return OutlineDecoration(
-      child: Column(
-        crossAxisAlignment: CrossAxisAlignment.end,
-        children: [
-          TimelineAnalysisHeader(
-            controller: _controller,
-            selectedTab: selectedTab,
-            searchFieldBuilder: () => _buildSearchField(searchFieldEnabled),
-          ),
-          Expanded(
-            child: Padding(
-              padding: const EdgeInsets.only(bottom: 8.0),
-              child: content,
-            ),
-          ),
-        ],
+    const rasterMetrics = Center(
+      child: Text('Coming Soon'),
+    );
+
+    final tabViews = [
+      TimelineEventsView(
+        controller: controller,
+        processing: widget.processing,
+        processingProgress: widget.processingProgress,
       ),
+      if (frameAnalysisSupported) frameAnalysisView,
+      if (rasterMetricsSupported) rasterMetrics,
+    ];
+
+    return AnalyticsTabbedView(
+      tabs: _generateTabs(),
+      tabViews: tabViews,
+      gaScreen: analytics_constants.performance,
+      // TODO(kenz): enable analytics when this view is stable.
+      sendAnalytics: false,
     );
   }
 
+  List<DevToolsTab> _generateTabs() {
+    final searchFieldEnabled =
+        !(controller.data?.isEmpty ?? true) && !widget.processing;
```

Comment:
Optional: pull `!(controller.data?.isEmpty ?? true)` into a variable like `hasData` to make this easier to read

Reply:
done